iam/src/providers/githubStarredRepoProvider.ts

```diff
@@ -0,0 +1,143 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+import type {
+  GithubFindMyUserResponse,
+  GithubRepoRequestResponse,
+  GithubUserRepoResponseData,
+} from "./types/githubTypes";
+
+// ----- HTTP Client
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+type StargazerData = {
+  id?: string | number;
+};
+
+// Export a Github Provider to carry out OAuth and return a record object
+export class StarredGithubRepoProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "StarredGithubRepoProvider";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      accessToken: string,
+      verifiedUserPayload: GithubFindMyUserResponse = {},
+      verifiedUserRepoPayload: boolean | GithubUserRepoResponseData = {};
+
+    try {
+      accessToken = await requestAccessToken(payload.proofs.code);
+      verifiedUserPayload = await verifyGithub(accessToken);
+      verifiedUserRepoPayload = await verifyUserGithubRepo(verifiedUserPayload, accessToken);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid = verifiedUserPayload && verifiedUserPayload.id && verifiedUserRepoPayload ? true : false;
+    }
```

Comment:
I would change this:

```suggestion
    try {
      accessToken = await requestAccessToken(payload.proofs.code);
      verifiedUserPayload = await verifyGithub(accessToken);
      verifiedUserRepoPayload = await verifyUserGithubRepo(verifiedUserPayload, accessToken);

      valid = verifiedUserPayload && verifiedUserPayload.id && verifiedUserRepoPayload ? true : false;
    } catch (e) {
      return { valid: false };
    } 
```